sycl/source/detail/queue_impl.hpp

```diff
@@ -769,14 +780,13 @@ class queue_impl {
 
     synchronizeWithExternalEvent(Handler);
 
-    auto Event = Handler.finalize();
-    EventToBuildDeps = getSyclObjImpl(Event);
-    assert(!EventToBuildDeps->isDiscarded());
-    return Event;
+    EventToBuildDeps = parseEvent(Handler.finalize());
```

Comment:
Can we save the assert from initial version? Something like  assert(EventToBuildDeps);